### PR TITLE
Unify implementations on computing size from shape

### PR DESCRIPTION
**Pull request by toslunar**

### CODE REVIEW

File: chainer/utils/array.py [hunk 1]
```diff
@@ -39,6 +41,10 @@ def empty_like(x):
         return numpy.empty_like(x)
 
 
+def size_of_shape(shape):
+    return functools.reduce(operator.mul, shape, 1)
```
Comment (okuta) [inline chainer/utils/array.py]: Is this first?
Cloud you check speed?
```
size = 1
for i in shape:
    size *= i
return size
```

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by kuenishi** [comment]
Good catch! A lot of same idioms occur at https://github.com/chainer/chainer/blob/master/chainermn/communicators/mpi_communicator_base.py , can you also apply this changes to them?

**Comment by toslunar** [comment]
@kuenishi Yes, I will.

**Comment by okuta** [comment]
OK, I will merge this PR.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**